### DISCUSSION TITLE
Coordinate system in the viewer?

### DISCUSSION THREAD

**Discussion by tve** [sgenoud/replicad#86] [Q&A]
I'm a bit confused about the intent of the coordinate system in the viewer. As far as I can tell, the API uses a coordinate system with Y pointing "up" and Z pointing into the scene (typ for computer graphics) but the viewer places objects onto the XY plane and so Z ends up pointing "up", at least the way I see it... This becomes very confusing when one uses the `width`, `depth` `height` accessors.
What is the thinking here? (I realize 3D printers generally print onto the XY plane.) Any chance one could toggle the visualizer between the two modes? Or am I completely missing something?

**Comment by sgenoud** [comment]
> the API uses a coordinate system with Y pointing "up" and Z pointing into the scene (typ for computer graphics)

What suggests this for you? I have worked with Z is up all the time and iterated on the API that way?

**Reply by raydeleu** [reply]
For me the current definition is fine (or stronger: preferable) as it conforms to the definition in most CAD and 3D tools (for example Blender) where x,y defines the horizontal plane and z defines the height. As mentioned, it is also the definition for 3D printers.

**Reply by tve** [reply]
I agree with Z being up being preferable. The reason I assumed that the API uses Y-up, Z-forward is the following:
```
  const box = r.makeBaseBox(10, 20, 30)
  const bb = box.boundingBox
  console.log(`width=${bb.width} depth=${bb.depth} height=${bb.height}`)
```
which prints:
```
width=10.0000002 depth=30.000000200000002 height=20.0000002
```
which is exactly what a std computer graphics Y-up coordinate system would call things...

**Comment by raydeleu** [comment]
I understand your confusion, I never had the need to assess the bounding box. In the Workbench it looks completely as expected. The names "width, depth and height" are indeed confusing. 

![image](https://github.com/sgenoud/replicad/assets/38007983/c046e503-7aaa-441d-a8e7-e09c557e3836)

**Comment by sgenoud** [comment]
Thanks, perhaps a little bit of background on why I call these dimensions like that.

I really started drawing with code for 2D stuff - and I used as a convention width and height for X and Y axes respectively. I guess this is not unexpected (is it?, English is not my first language). 

The next step is when I went into 3D I needed a new term for the Z axis - so it became depth. I use it as a convention for myself and forgot this is my own convention.

I should probably call the `xLength`, `yLength` and `zLength` to make it more explicit.
